lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -0,0 +1,580 @@
+    // in the referenced module, to the children.
+    //
+    if (!node.isa<BlockArgument>()) {
+      // Any Value that is derived from a Register must be ignored. `skip` is
+      // true, only if this node is the result of some Register.
+      bool skip =
+          TypeSwitch<Operation *, bool>(node.getDefiningOp())
+              .Case<InstanceOp>([&](InstanceOp ins) {
+                size_t portNum = 0;
+                // Get the port index for `node`.
+                for (const auto &p : llvm::enumerate(ins->getOpResults()))
+                  if (p.value() == node) {
+                    portNum = p.index();
+                    break;
+                  }
+                FModuleOp referencedModule = dyn_cast<FModuleOp>(
+                    *instanceGraph.getReferencedModule(ins));
+                if (referencedModule) {
+                  auto port =
+                      referencedModule.getBodyBlock()->getArgument(portNum);
+                  // Get all the ports that have a comb path from `port`.
+                  for (const auto &portNode : portPaths[port])
+                    for (auto outputPort : portNode.getSecond()) {
+                      auto instResult = ins.getResult(
+                          outputPort.val.cast<BlockArgument>().getArgNumber());
+                      // Note that, in this case the instResult is always of
+                      // Ground Type. FieldRefs are handled as a pre-pass before
+                      // the DFS begins.
+                      children.push_back({instResult, ins});
+                    }
+                }
+                return false;
+                ;
+              })
+              .Case<RegOp, RegResetOp>([&](auto) {
+                children.clear();
+                return true;
+              })
+              .Case<SubfieldOp, SubindexOp, SubaccessOp>([&](auto sub) {
+                // If the subfield result is not recorded then it must be
+                // ignored. Only aggregate port and wire ops are recorded.
+                if (valToStorageLocIdMap.find(node) ==
+                    valToStorageLocIdMap.end()) {
+                  children.clear();
+                  return true;
+                }
+                return false;
+                ;
+              })
+              .Default([&](auto) { return false; });
+      if (skip)
+        return;
+    }
+
+    // Now, handle the ops which create an explicit comb path from `node` to
+    // another Value.
+    for (auto &child : node.getUses()) {
+      Operation *owner = child.getOwner();
+      LLVM_DEBUG(llvm::dbgs() << "\n owner: " << *owner);
+      auto childVal =
+          TypeSwitch<Operation *, Value>(owner)
+              .Case<FConnectLike>([&](FConnectLike connect) {
+                // Ignore connects, where the child is the destination.
+                if (child.getOperandNumber() == 0)
+                  return (Value) nullptr;
```

Comment:
I think `{}` looks more consistent to other places
```suggestion
                  return {};
```

Reply:
If explicit cast is required, does it work to specify the return type for anonymous function ? `[&](FConnectLike connect) -> Value`?

Reply:
The explicit return is a better idea. Updated.